src/main/java/com/zaxxer/sansorm/internal/AttributeInfo.java

```diff
@@ -10,59 +10,66 @@
 /**
  * Column information about a field
  */
-final class FieldColumnInfo
+abstract class AttributeInfo
 {
-   private final Class<?> clazz;
-   private final String propertyName;
+   protected final Class<?> clazz;
+   protected String name;
 
    final Field field;
-   final Class<?> fieldType;
+   Class<?> type;
 
-   private boolean isDelimited;
-   private Boolean updatable;
-   private Boolean insertable;
-   private String columnName;
-   /** name without delimiter: lower cased; delimited name: name as is with delimiters */
+   protected boolean isDelimited;
+   protected Boolean updatable;
+   protected Boolean insertable;
+   protected String columnName;
+   /** name without delimiter: name as is; delimited name: name as is with delimiters */
    String columnTableName = "";
    EnumType enumType;
    Map<Object, Object> enumConstants;
-   private AttributeConverter converter;
-   private String caseSensitiveColumnName;
-   boolean isGeneratedId;
+   protected AttributeConverter converter;
+   protected String caseSensitiveColumnName;
+   protected boolean isGeneratedId;
    boolean isIdField;
-   private boolean isJoinColumn;
-   boolean isTransient;
-   private boolean isEnumerated;
-   private boolean isColumnAnnotated;
-   private String delimitedFieldName;
-   private final String fullyQualifiedDelimitedFieldName;
-
-   public FieldColumnInfo(Field field, Class<?> clazz) {
+   protected boolean isJoinColumn;
+   protected boolean isTransient;
+   protected boolean isEnumerated;
+   protected boolean isColumnAnnotated;
+   protected String delimitedName;
+   protected final String fullyQualifiedDelimitedName;
+
+   public AttributeInfo(Field field, Class<?> clazz) {
       this.field = field;
       this.clazz = clazz;
-      this.propertyName = field.getName();
-      this.fieldType = getFieldType();
+      extractFieldName(field);
+      adjustType(extractType());
       extractAnnotations();
       processFieldAnnotations();
-      this.fullyQualifiedDelimitedFieldName =
-         columnTableName.isEmpty() ? delimitedFieldName : columnTableName + "." + delimitedFieldName;
+      this.fullyQualifiedDelimitedName =
+         columnTableName.isEmpty() ? delimitedName : columnTableName + "." + delimitedName;
    }
 
-   private Class<?> getFieldType() {
-      final Class<?> type = field.getType();
+   protected abstract void extractFieldName(Field field);
```

Comment:
It might seem nit-picky, but I'm all about using ``final`` on [parameters](https://github.com/brettwooldridge/SansOrm/blob/master/src/main/java/com/zaxxer/sansorm/internal/Introspected.java#L186) and [locals](https://github.com/brettwooldridge/SansOrm/blob/master/src/main/java/com/zaxxer/sansorm/internal/Introspected.java#L403).  Inadvertent modification can cause difficult to uncover issues and allows the IDE and compiler to catch such issues.

Reply:
I have now activated "Make generated local variables final" and "Make generated parameters final" in IntelliJ preferences.